### PR TITLE
Initial commit for HAML cheatsheet

### PR DESCRIPTION
**Pull request by slaterio**
###### What does your Pull Request do (check all that apply)?

Choose the most relevant items and use the following title template to name
your Pull Request.

- [x] New Instant Answer
    - [x] Cheat Sheets: **`New HAML Cheat Sheet`**
    - [ ] Other: **`New {IA Name} Instant Answer`**
- [ ] Improvement
    - [ ] Bug fix: **`{IA Name}: Fix {Issue number or one-line description}`**
    - [ ] Enhancement: **`{IA Name}: {Description of Improvements}`**
- [ ] Non–Instant Answer
    - [ ] Other (Role, Template, Test, Documentation, etc.): **`{GoodieRole/Templates/Tests/Docs}: {Short Description}`**

###### Description of changes

Initial commit for HAML cheatsheet.

---

Instant Answer Page: https://duck.co/ia/view/haml_cheat_sheet

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @mention

### CODE REVIEW

**Comment by daxtheduck** [comment]
@slaterio Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please head to https://quackslack.herokuapp.com/ to get an invite.

>More Info: http://docs.duckduckhack.com/submitting/submitting-overview.html

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/goodie/cheat_sheets/json/haml.json [hunk 1]
```diff
@@ -0,0 +1,154 @@
+{
+	"id": "haml_cheat_sheet",
+	"name": "HAMLCheatsheet",
```
Comment (tagawa) [inline share/goodie/cheat_sheets/json/haml.json]: Please could this be changed to "HAML Cheat Sheet"?

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=HAML Cheat Sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/haml_cheat_sheet_mobile.png?FjUViDOeFQ)

**Comment by tagawa** [comment]
@slaterio Thanks for sending this pull request with the changed filename. I forced a re-check of Travis and it worked, with the cheat sheet now installed on the beta server: https://beta.duckduckgo.com/?q=HAML+Cheat+Sheet&ia=cheatsheet&iax=1

Looking good so far, although there are a few errant characters here and there. This should be easier to troubleshoot now that we can see it live.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=HAML Cheat Sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/haml_cheat_sheet_mobile.png?JuXG9SIoLH)

File: share/goodie/cheat_sheets/json/haml.json [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+	"template_type": "terminal",
+
+	"section_order": [
+		"Basics",
+		"HTML Elements",
+        "Comments",
+        "Doctypes",
+        "Ruby Evaluation",
+        "Filters"
+	],
```
Comment (tagawa) [inline share/goodie/cheat_sheets/json/haml.json]: Please could these four be indented to match the first two.

File: share/goodie/cheat_sheets/json/haml.json [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+			"key": "%a{ :href => 'https://duckduckgo.com/' }",
+			"val": "<a href='https://duckduckgo.com/'>"
+		}],
+
+		"Comments": [{
+			"key": "/ This is the peanutbutterjelly element",
+			"val": "<!-- This is the peanutbutterjelly element -->"
+		}, {
+			"key": "/[if IE]",
+			"val": "<!--[if IE]>"
```
Comment (tagawa) [inline share/goodie/cheat_sheets/json/haml.json]: To keep the square brackers, please replace this key with `"/\\[if IE\\]"`

File: share/goodie/cheat_sheets/json/haml.json [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+				"key": "!!! Frameset",
+				"val": "<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Frameset//EN\" \"http://www.w3.org/TR/xhtml1/DTD/xhtml1-frameset.dtd\">"
+			}
+		],
+
+		"Ruby Evaluation": [
+
+			{
+				"key": "= ['hi', 'there', 'reader!'].join \"\"",
+				"val": "hi there reader!"
```
Comment (tagawa) [inline share/goodie/cheat_sheets/json/haml.json]: To keep the correct formatting, please replace the key here with `"= \\['hi', 'there', 'reader!'\\].join \" \""`

File: share/goodie/cheat_sheets/json/haml.json [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+			"val": "<br />"
+		}, {
+			"key": "%div#things",
+			"val": "<div id='things'>"
+		}, {
+			"key": "%span#rice Chicken Fried",
+			"val": "<span id='rice'>Chicken Fried</span>"
+		}, {
+			"key": "%p.beans{ :food => 'true' } The magical fruit",
+			"val": "<p class='beans' food='true'>The magical fruit</p>"
```
Comment (tagawa) [inline share/goodie/cheat_sheets/json/haml.json]: To keep the formatting here, please change the key to 
`"%p.beans\\{ :food => 'true' \\} The magical fruit"`

File: share/goodie/cheat_sheets/json/haml.json [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+		}, {
+			"key": "%h1.class.otherclass#id La La La",
+			"val": "<h1 class='class otherclass' id='id'>La La La</h1>"
+		}, {
+			"key": "%a{ :href => 'https://duckduckgo.com/' }",
```
Comment (tagawa) [inline share/goodie/cheat_sheets/json/haml.json]: To fix the formatting, please change this key to 
`"%a\\{ :href => 'https://duckduckgo.com/' \\}"`

**Comment by tagawa** [comment]
Thanks for fixing the title @slaterio. There are still a few errant characters as indicated below. I've put some comments in the code which should fix them but I'm not familiar with HAML so you might want to double-check.

haml-cheatsheet

**Comment by tagawa** [comment]
Thank you @slaterio. Any chance you could indent lines 20-23? That should be all I think.

**Comment by slaterio** [comment]
I have those lines indented in the file. All lines from 18-23 are indented using two tabs.
haml-indent01

**Comment by tagawa** [comment]
@slaterio Ah, I see - it must be the formatting in GitHub. I'm installing this to the beta server now for testing, and if there are no further comments we can merge this soon.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=HAML Cheat Sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/haml_cheat_sheet_mobile.png?xNY9aCD9kq)

**Comment by tagawa** [comment]
@slaterio Well no further feedback so let's merge this. This is a really handy cheat sheet for web developers - thank you very much!

**Comment by daxtheduck** [comment]
@slaterio this is now merged! It should be live on DuckDuckGo within one week (or during the next release) and we'll be sure to notify you when it happens.

After your Instant Answer goes live for the World to use, you will be able to see [traffic information](http://docs.duckduckhack.com/maintaining/tools.html#monitor-your-instant-answer-page) about your IA as well as receive feedback directly from users. 
                
Please refer to the [maintenance guide and docs](http://docs.duckduckhack.com/maintaining/guidelines.html) to keep up with potential changes to your Instant Answer.
                
Thank you again for your help!

**Pull request merged**